tests/features/mocking.py

```diff
@@ -804,6 +804,34 @@ def method(cls, arg):
         flexmock(instance).should_receive("method").with_args("stuff")
         instance.method("stuff")
 
+    def test_with_args_scoped_mock(self):
+        class FooClass:
+            def method(self, arg):
+                return arg
+
+        flexmock(FooClass).should_receive("method").with_args("override").and_return("method override").once()
+        instance = FooClass()
+        assert instance.method("override") == "method override"
+        assert instance.method("original") == "original"
+
+    def test_with_args_on_new_constructor(self):
+        class FooClass:
+            def __init__(self, arg):
+                self.arg = arg
+
+            def method(self):
+                return self.arg
+
+        fake_Foo = flexmock()
+        fake_Foo.should_receive("method").and_return("method override")
+        flexmock(FooClass).should_receive("__new__").with_args("override").and_return(fake_Foo)
```

Comment:
This one is a little trickier because we're playing with `__new__` but the same pattern can be applied
```suggestion
        flexmock(FooClass).should_receive("__new__").replace_with(lambda x: fake_Foo if x == "override" else object.__new__(FooClass))
```